search-service/src/main/kotlin/com/egm/stellio/search/temporal/model/TemporalEntitiesQuery.kt

```diff
@@ -1,31 +1,55 @@
 
 import com.egm.stellio.search.entity.model.EntitiesQuery
+import com.egm.stellio.search.entity.model.EntitiesQueryFromGet
+import com.egm.stellio.search.entity.model.EntitiesQueryFromPost
 import java.time.Duration
 import java.time.Period
 import java.time.temporal.TemporalAmount
 
-data class TemporalEntitiesQuery(
-    val entitiesQuery: EntitiesQuery,
-    val temporalQuery: TemporalQuery,
-    val withTemporalValues: Boolean,
-    val withAudit: Boolean,
-    val withAggregatedValues: Boolean
+sealed class TemporalEntitiesQuery(
+    open val temporalQuery: TemporalQuery,
+    open val withTemporalValues: Boolean,
+    open val withAudit: Boolean,
+    open val withAggregatedValues: Boolean
 ) {
     fun isAggregatedWithDefinedDuration(): Boolean =
         withAggregatedValues &&
             temporalQuery.aggrPeriodDuration != null &&
             temporalQuery.aggrPeriodDuration != "PT0S"
 
-    fun computeAggrPeriodDuration(): TemporalAmount {
-        val splitted = temporalQuery.aggrPeriodDuration!!.split("T")
-        return if (splitted.size == 1) // has only date-based fields
-            Period.parse(temporalQuery.aggrPeriodDuration)
-        else {
-            val duration = Duration.parse("PT" + splitted[1])
-            if ("P" == splitted[0]) // has only time-based fields
-                duration
-            else Period.parse(splitted[0]).plus(duration)
-        }
-    }
+    fun computeAggrPeriodDuration(): TemporalAmount =
+        temporalQuery.aggrPeriodDuration?.let { aggrPeriodDuration ->
+            val splitted = aggrPeriodDuration.split("T")
+            if (splitted.size == 1) // has only date-based fields
+                Period.parse(aggrPeriodDuration)
+            else {
+                val duration = Duration.parse("PT" + splitted[1])
+                if ("P" == splitted[0]) // has only time-based fields
+                    duration
+                else Period.parse(splitted[0]).plus(duration)
+            }
+        } ?: Period.ZERO
+
+    abstract fun getEntitiesQuery(): EntitiesQuery
+}
+
+data class TemporalEntitiesQueryFromGet(
+    val entitiesQueryFromGet: EntitiesQueryFromGet,
+    override val temporalQuery: TemporalQuery,
+    override val withTemporalValues: Boolean,
+    override val withAudit: Boolean,
+    override val withAggregatedValues: Boolean
+) : TemporalEntitiesQuery(temporalQuery, withTemporalValues, withAudit, withAggregatedValues) {
+    override fun getEntitiesQuery(): EntitiesQuery = entitiesQueryFromGet
+}
+
+data class TemporalEntitiesQueryFromPost(
+    val entitiesQueryFromPost: EntitiesQueryFromPost,
+    override val temporalQuery: TemporalQuery,
+    override val withTemporalValues: Boolean,
+    override val withAudit: Boolean,
+    override val withAggregatedValues: Boolean
+) : TemporalEntitiesQuery(temporalQuery, withTemporalValues, withAudit, withAggregatedValues) {
+    override fun getEntitiesQuery(): EntitiesQuery = entitiesQueryFromPost
```

Comment:
I think you can avoid deleting :
```
val entitiesQuery: EntitiesQuery
```
from TemporalEntitiesQuery.

 Then, TemporalEntitiesQueryFromPost and TemporalEntitiesQueryFromGet could override it with their respective types.

This approach would eliminate the need to create and call getEntitiesQuery().

Reply:
indeed, better. done.